File: FrontEnd/src/app/Class/songs.ts
import { Bands } from './bands'

export class Songs{

	private id: number;
	private songName: String;
	private yearOfCreation: number;
	private genre: String;
	private lengthOfSong: number;
	private averageRateOfSong: number;
	private albums: String;
	private band: String;


	constructor(obj?: any){
	
		this.id = (obj && obj.id) || 0;
		this.songName = (obj && obj.songName) || '';
		this.yearOfCreation = (obj && obj.yearOfCreation) || 0;
		this.genre = (obj && obj.genre) || '';
		this.lengthOfSong = (obj && obj.lengthOfSong) || 0;
		this.averageRateOfSong = (obj && obj.averageRateOfSong) || 0;
		this.albums = (obj && obj.albums) || '';
		this.band = (obj && obj.band) || '';

	}

	getId(): number{
		return this.id;
	}
	setId(id: number){
		this.id = id;
	}
	
	getSongName(): String{
		return this.songName;
	}
	setSongName(songName: String){
		this.songName = songName;
	}
	
	getYearOfCreation(): number{
		return this.yearOfCreation;
	}
	setYearOfCreation(yearOfCreation: number){
		this.yearOfCreation = yearOfCreation;
	}
	
	getGenre(): String{
		return this.genre;
	}
	setGenre(genre: String){
		this.genre = genre;
	}
	
	getLengthOfSong(): number{
		return this.lengthOfSong;
	}
	setLengthOfSong(lengthOfSong: number){
		this.lengthOfSong = lengthOfSong;
	}

	getAverageRateOfAlbum(): number{
		return this.averageRateOfSong;
	}
	setAverageRateOfAlbum(averageRateOfSong: number){
		this.averageRateOfSong = averageRateOfSong;
	}

	getAlbums(): String{
		return this.albums;
	}
	setAlbums(albums: String){
		this.albums = albums;
	}

	getBand(): String{
		return this.band;
	}
	setBand(band: String){
		this.band = band;
	}


}

File: Backend/src/main/java/Spring/Entities/Songs.java
package Spring.Entities;

import javax.persistence.*;

@Entity
@Table(name = "Songs", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"id"})
})
public class Songs {

    @Id
    @GeneratedValue
    @Column(name = "id")
    private int id;

    @Column(name = "Song_Name")
    private String songName;

    @Column(name = "Year_of_creation")
    private int yearOfCreation;

    @Column(name = "Genre")
    private String genre;

    @Column(name = "Lenght_Of_Song")
    private int lengthOfSong;

    @Column(name = "Averange_Rate_Of_Song")
    private float averangeRateOfSong;

    @Column(name = "Albums")
    private String albums;


    @Column(name = "Band")
    private String band;

    public Songs() {
    }

    public Songs(String songName, int yearOfCreation, String genre, int lengthOfSong, float averangeRateOfSong) {
        this.songName = songName;
        this.yearOfCreation = yearOfCreation;
        this.genre = genre;
        this.lengthOfSong = lengthOfSong;
        this.averangeRateOfSong = averangeRateOfSong;
        //this.albums = albums;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getSongName() {
        return songName;
    }

    public void setSongName(String songName) {
        this.songName = songName;
    }

    public int getYearOfCreation() {
        return yearOfCreation;
    }

    public void setYearOfCreation(int yearOfCreation) {
        this.yearOfCreation = yearOfCreation;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public int getLengthOfSong() {
        return lengthOfSong;
    }

    public void setLengthOfSong(int lengthOfSong) {
        this.lengthOfSong = lengthOfSong;
    }

    public float getAverangeRateOfSong() {
        return averangeRateOfSong;
    }

    public void setAverangeRateOfSong(float averangeRateOfSong) {
        this.averangeRateOfSong = averangeRateOfSong;
    }

    public String getAlbums() {
        return albums;
    }

    public void setAlbums(String albums) {
        this.albums = albums;
    }



}
